sdk/purview/purview-scanning-rest/src/purviewScanning.ts

```diff
@@ -1,489 +1,65 @@
-  (path: "/systemScanRulesets"): SystemScanRulesetsListAll;
-  /** Resource for '/systemScanRulesets/datasources/\{dataSourceType\}' has methods for the following verbs: get */
-  (
-    path: "/systemScanRulesets/datasources/{dataSourceType}",
-    dataSourceType: string,
-  ): SystemScanRulesetsGet;
-  /** Resource for '/systemScanRulesets/versions/\{version\}' has methods for the following verbs: get */
-  (path: "/systemScanRulesets/versions/{version}", version: string): SystemScanRulesetsGetByVersion;
-  /** Resource for '/systemScanRulesets/versions/latest' has methods for the following verbs: get */
-  (path: "/systemScanRulesets/versions/latest"): SystemScanRulesetsGetLatest;
-  /** Resource for '/systemScanRulesets/versions' has methods for the following verbs: get */
-  (path: "/systemScanRulesets/versions"): SystemScanRulesetsListVersionsByDataSource;
-  /** Resource for '/datasources/\{dataSourceName\}/scans/\{scanName\}/triggers/default' has methods for the following verbs: get, put, delete */
-  (
-    path: "/datasources/{dataSourceName}/scans/{scanName}/triggers/default",
-    dataSourceName: string,
-    scanName: string,
-  ): TriggersGetTrigger;
-}
-
-export type PurviewScanningRestClient = Client & {
-  path: Routes;
-};
-
-export default function PurviewScanning(
-  Endpoint: string,
+import type { PurviewScanningClient } from "./clientDefinitions";
+
+/** The optional parameters for the client */
+export interface PurviewScanningClientOptions extends ClientOptions {
+  /** The api version option of the client */
+  apiVersion?: string;
+}
+
+/**
+ * Initialize a new instance of `PurviewScanningClient`
+ * @param endpoint - The scanning endpoint of your purview account. Example: https://\{accountName\}.scan.purview.azure.com
+ * @param credentials - uniquely identify client credential
+ * @param options - the parameter for all optional parameters
+ */
+export default function createClient(
+  endpoint: string,
   credentials: TokenCredential,
-  options: ClientOptions = {},
-): PurviewScanningRestClient {
-  const baseUrl = options.baseUrl ?? `${Endpoint}`;
-  options.apiVersion = options.apiVersion ?? "2018-12-01-preview";
+  { apiVersion = "2018-12-01-preview", ...options }: PurviewScanningClientOptions = {},
+): PurviewScanningClient {
+  const endpointUrl = options.endpoint ?? options.baseUrl ?? `${endpoint}`;
+  const userAgentInfo = `azsdk-js-purview-scanning-rest/1.0.0-beta.3`;
+  const userAgentPrefix =
+    options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+      ? `${options.userAgentOptions.userAgentPrefix} ${userAgentInfo}`
+      : `${userAgentInfo}`;
   options = {
     ...options,
+    userAgentOptions: {
+      userAgentPrefix,
+    },
+    loggingOptions: {
+      logger: options.loggingOptions?.logger ?? logger.info,
+    },
     credentials: {
-      scopes: ["https://purview.azure.net/.default"],
+      scopes: options.credentials?.scopes ?? ["https://management.azure.com/.default"],
```

Comment:
have we set the openapi-type: data-plane in this case? I don't think we should set the credential scope as mgmt plane scope.